src/login/LoginPage.jsx

```diff
@@ -74,10 +99,29 @@ const LoginPage = (props) => {
 
   useEffect(() => {
     if (loginErrorCode) {
-      setErrorCode(prevState => ({ type: loginErrorCode, count: prevState.count + 1 }));
+      setErrorCode(prevState => ({ type: loginErrorCode, count: prevState.count + 1, 
+        context: {
+          errorMessage: loginErrorCode,
+      } }));
```

Comment:
i still have a feeling that we should not do like this. i know you have put effort on this, but this approach is a bit complicated IMO. 
I will try to come up with a better approach in a while if i could. otherwise we will go with this.

Reply:
Besides the approach there is a bug in the code too.

```suggestion
      setErrorCode(prevState => ({
          type: loginErrorCode,
          context: { ... loginErrorContext },
          count: prevState.count + 1, 
      }));
```

Reply:
addressed